packit/api.py

```diff
@@ -282,7 +282,10 @@ def sync_release(
             if not use_local_content:
                 self.up.local_project.git_repo.git.checkout(current_up_branch)
             self.dg.refresh_specfile()
-
+            branch = local_pr_branch if create_pr else dist_git_branch
+            self.dg.local_project.git_repo.git.reset(
+                "--hard", f"remotes/origin/{branch}"
+            )
```

Comment:
Will this work as well, or we need to really checkout the remote branch?

```suggestion
            self.dg.local_project.git_repo.git.reset("--hard", "HEAD")
```

Reply:
tests are passing with this